### PR TITLE
Added WiX detection to MSI Parser.

### PR DESCRIPTION
**Pull request by jedieaston**
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Are you working against an Issue?

-----
Resolves https://github.com/microsoft/winget-create/issues/222.

This PR adds WiX detection to the MSI Parser, inspired by similar changes to [YamlCreate.ps1](https://github.com/Trenly/winget-pkgs/pull/140/). Unfortuately, since there isn't a silver bullet way to determine if a MSI was created by WiX several things are checked.

Also, it seems to reformat PackageParser.cs for some reason, I suppose there was some style issue that was corrected during commit.

Tested: manually. As always, I'm happy to write unit tests if necessary.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-create/pull/220)

### CODE REVIEW

File: src/WingetCreateCore/Common/PackageParser.cs [hunk 1]
```diff
@@ -643,6 +643,37 @@ private static bool ParseExeInstallerType(string path, Installer baseInstaller,
...
+            }
+
+            // Do any of the WiX tables exist?
+            foreach (var table in installer.Tables)
+            {
+                if (table.Name.ToLower().Contains("wix"))
+                {
+                    return true;
+                }
+            }
```
Comment (Trenly) [inline src/WingetCreateCore/Common/PackageParser.cs:669]:
```suggestion
            if Array.Exists(installer.Tables, table => table.Name.ToLower().Contains("wix")) return true;
```

File: src/WingetCreateCore/Common/PackageParser.cs [hunk 2]
```diff
@@ -643,6 +643,37 @@ private static bool ParseExeInstallerType(string path, Installer baseInstaller,
...
+        private static bool IsWix(QDatabase installer)
+        {
+            // Do any of the WiX properties exist?
+            foreach (var property in installer.Properties)
+            {
+                if (property.Property.ToLower().Contains("wix"))
+                {
+                    return true;
+                }
+            }
```
Comment (Trenly) [inline src/WingetCreateCore/Common/PackageParser.cs:660]:
```suggestion
            if Array.Exists( installer.Properties, property => property.Property.ToLower().Contains("wix")) return true;
```

Comment (Trenly) [reply src/WingetCreateCore/Common/PackageParser.cs:660]: Note that you could remove the `if` and `return true` here by or-ing them with your other two statements

File: src/WingetCreateCore/Common/PackageParser.cs [hunk 3]
```diff
@@ -643,6 +643,37 @@ private static bool ParseExeInstallerType(string path, Installer baseInstaller,
...
+                if (table.Name.ToLower().Contains("wix"))
+                {
+                    return true;
+                }
+            }
+
+            // Does the CreatingApp mention WiX?
+            return
+                installer.SummaryInfo.CreatingApp.ToLower().Contains("wix") ||
+                installer.SummaryInfo.CreatingApp.ToLower().Contains("windows installer xml");
```
Comment (Trenly) [inline src/WingetCreateCore/Common/PackageParser.cs:674]:
```suggestion
            // Do any of the WiX properties exist?
            // Do any of the WiX tables exist?
            // Does the CreatingApp mention WiX?
            return
                Array.Exists(installer.Properties, property => property.Property.ToLower().Contains("wix")) ||
                Array.Exists(installer.Tables, table => table.Name.ToLower().Contains("wix")) ||
                installer.SummaryInfo.CreatingApp.ToLower().Contains("wix") ||
                installer.SummaryInfo.CreatingApp.ToLower().Contains("windows installer xml");
```

Comment (jedieaston) [reply src/WingetCreateCore/Common/PackageParser.cs]: Had to modify your suggestions to convert them to Arrays (they didn't have Exist methods and apparently you can't just do IEnumerable.Exists on them either). Want to take a look?

Comment (Trenly) [reply src/WingetCreateCore/Common/PackageParser.cs]: Replied in Gitter; May be able to use the `FirstOrDefault` method instead

**Review by Trenly — APPROVED** [review]

**Comment by jedieaston** [comment]
Component Governance down again? 

```
Starting: NOTICE File Generator
==============================================================================
Task         : OSPO NOTICE File Generator
Description  : Generates a NOTICE file based on a snapshot
Version      : 0.5.14
Author       : ospo
Help         : Generates a NOTICE file based on a Component Governance Snapshot
==============================================================================
##[error]failed to get governed repository id
##[error]Generation failed. Please see https://aka.ms/noticedebugging for more information.
Finishing: NOTICE File Generator

```

**Review by ryfu-msft — APPROVED** [review]
:shipit:

**Comment by ryfu-msft** [comment]
@jedieaston, can you file an issue relating to this change? Just want to make sure that every PR has a linked issue for tracking purposes. In the meantime, I'm looking into why the build is failing...

**Comment by Trenly** [comment]
> @jedieaston, can you file an issue relating to this change? Just want to make sure that every PR has a linked issue for tracking purposes. In the meantime, I'm looking into why the build is failing...

#222

**Comment by ryfu-msft** [comment]
@jedieaston, sorry for the delay, took some time to figure out what was going on with Component Governance. You should be good to go now :)

**Comment by jedieaston** [comment]
Looks like the changes from main are merged. Ready when you are...

**Pull request merged**